python_modules/libraries/dagster-databricks/dagster_databricks/ext.py

```diff
@@ -0,0 +1,156 @@
+    )
+
+    def __init__(self, client: WorkspaceClient, env: Optional[Mapping[str, str]] = None):
+        self.client = client
+        self.env = env
+
+    def run(
+        self,
+        task: jobs.SubmitTask,
+        *,
+        context: OpExecutionContext,
+        extras: Optional[ExtExtras] = None,
+        context_injector: Optional[ExtContextInjector] = None,
+        message_reader: Optional[ExtMessageReader] = None,
+        submit_args: Optional[Mapping[str, str]] = None,
+    ) -> None:
+        """Run a Databricks job with the EXT protocol.
+
+        Args:
+            task (databricks.sdk.service.jobs.SubmitTask): Specification of the databricks
+                task to run. Environment variables used by dagster-ext will be set under the
+                `spark_env_vars` key of the `new_cluster` field (if there is an existing dictionary
+                here, the EXT environment variables will be merged in). Everything else will be
+                passed unaltered under the `tasks` arg to `WorkspaceClient.jobs.submit`.
+            submit_args (Optional[Mapping[str, str]]): Additional keyword arguments that will be
+                forwarded as-is to `WorkspaceClient.jobs.submit`.
+        """
+        with ext_protocol(
+            context=context,
+            extras=extras,
+            context_injector=context_injector or ExtDbfsContextInjector(client=self.client),
+            message_reader=message_reader or ExtDbfsMessageReader(client=self.client),
+        ) as ext_context:
+            submit_task_dict = task.as_dict()
+            submit_task_dict["new_cluster"]["spark_env_vars"] = {
+                **submit_task_dict["new_cluster"].get("spark_env_vars", {}),
+                **(self.env or {}),
+                **ext_context.get_external_process_env_vars(),
+            }
+            task = jobs.SubmitTask.from_dict(submit_task_dict)
+            run_id = self.client.jobs.submit(
+                tasks=[task],
+                **(submit_args or {}),
+            ).bind()["run_id"]
+
+            while True:
+                run = self.client.jobs.get_run(run_id)
+                if run.state.life_cycle_state in (
+                    jobs.RunLifeCycleState.TERMINATED,
+                    jobs.RunLifeCycleState.SKIPPED,
+                ):
+                    if run.state.result_state == jobs.RunResultState.SUCCESS:
+                        return
+                    else:
+                        raise DagsterExternalExecutionError(
+                            f"Error running Databricks job: {run.state.state_message}"
+                        )
+                elif run.state.life_cycle_state == jobs.RunLifeCycleState.INTERNAL_ERROR:
+                    raise DagsterExternalExecutionError(
+                        f"Error running Databricks job: {run.state.state_message}"
+                    )
+                time.sleep(5)
```

Comment:
I think we should log on every tick so the user is confident something is happening